arithmetization/src/main/java/net/consensys/linea/blockcapture/snapshots/StorageSnapshot.java

```diff
@@ -19,17 +19,36 @@
 
 import org.apache.tuweni.units.bigints.UInt256;
 import org.hyperledger.besu.datatypes.Address;
+import org.hyperledger.besu.evm.account.Account;
 import org.hyperledger.besu.evm.worldstate.WorldView;
 
 public record StorageSnapshot(String address, String key, String value) {
   public static Optional<StorageSnapshot> from(
       Address address, UInt256 key, final WorldView world) {
-    return Optional.ofNullable(world.get(address))
-        .map(
-            account ->
-                new StorageSnapshot(
-                    address.toHexString(),
-                    key.toHexString(),
-                    account.getStorageValue(key).toHexString()));
+    // Lookup account that was touched
+    Account account = world.get(address);
+    // Check account *really* exists from a storage perspective.  This is important to distinguish
```

Comment:
So in the EYP lingo account existence is equivalent to satisfying at least one of the three conditions below:
- nonzero balance
- nonzero nonce
- nonempty code

Storage is notably absent from that definition.